ruler-frontend/src/main/kotlin/com/spotify/ruler/frontend/components/Common.kt

```diff
@@ -44,13 +44,13 @@ import react.useState
 fun RBuilder.report(report: AppReport) {
     val (sizeType, setSizeType) = useState(Measurable.SizeType.DOWNLOAD)
 
-    val hasOwnershipInfo = report.components.any { component -> component.owner != null }
+    val hasOwnershipInfo = report.ownershipOverview?.isNotEmpty() == true
     val hasDynamicFeatures = report.dynamicFeatures.isNotEmpty()
 
     val tabs = listOf(
         Tab("/", "Breakdown") { breakdown(report.components, sizeType) },
-        Tab("/insights", "Insights") { insights(report.components) },
-        Tab("/ownership", "Ownership", hasOwnershipInfo) { ownership(report.components, sizeType) },
+        Tab("/insights", "Insights") { insights(report.insights) },
+        Tab("/ownership", "Ownership", hasOwnershipInfo) { ownership(report.components, sizeType, report.ownershipOverview ?: emptyMap()) },
```

Comment:
If we propagate already calculated data to the frontend, it doesn't have to iterate over files to obtain the insights and ownership overview, that way we can get rid of files listing if needed